sdk/data/aztables/client.go

```diff
@@ -652,6 +652,7 @@ func (s *SetAccessPolicyOptions) toGenerated() *generated.TableClientSetAccessPo
 // If the service returns a non-successful HTTP status code, the function returns an *azcore.ResponseError type.
 // Specify nil for options if you want to use the default options.
 func (t *Client) SetAccessPolicy(ctx context.Context, options *SetAccessPolicyOptions) (SetAccessPolicyResponse, error) {
+	if options == nil {options = &SetAccessPolicyOptions{}}
 	response, err := t.client.SetAccessPolicy(ctx, t.name, generated.Enum4ACL, options.toGenerated())
 	if len(options.TableACL) > 5 {
```

Comment:
Should this check be moved before the call to the internal client?

Reply:
That depends on if we want to do client side validation. We don't do this before sending to the service in other languages so I kept the pattern in Go. It also prevents us from having to issue a fix if the service were to accept more than 5 policies in a request.